src/libs/elektra/kdb.c

```diff
@@ -693,6 +726,15 @@ int kdbGet (KDB * handle, KeySet * ks, Key * parentKey)
 
 	ksRewind (ks);
 
+	if (elektraGetDoUpdate2 (split, ks, parentKey) == -1)
```

Comment:
I think the problem is: this is called too late, you cannot call doUpdate after postprocessing of the keysets.

why not:
```
elektraGetDoUpdate
hook MIDDLEGETSTORAGE
elektraGetDoUpdate2
elektraSplitGet // postprocessing
...
hook POSTGETSTORAGE
```

and spec will be called in MIDDLEGETSTORAGE